ggshield/secret/repo.py

```diff
@@ -192,6 +193,13 @@ def scan_commit_range(
                         ignored_detectors,
                     )
                 )
+                # Stop now if an exception has been raised by a future
+                if any(future.exception() is not None for future in futures):
+                    raise next(
+                        future.exception()
+                        for future in futures
+                        if future.exception() is not None
+                    )
```

Comment:
Isn't it too early to check for the exception? I would have expected the call to `future.result()` in the loop below to be enough.

Reply:
We can win a lot of time by doing it here. Here's the different on ggshield repo:
```shell
# Without
time ggshield secret scan repo .
Scanning Commits... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 1243 commits scanned out of 1243 0:00:00

Error: Could not perform the requested action: no more API calls available.
ggshield secret scan repo .  7.31s user 9.87s system 78% cpu 22.028 total

# With
$ time ggshield secret scan repo .
Scanning Commits... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% 1 commits scanned out of 1243 -:--:--

Error: Could not perform the requested action: no more API calls available.
ggshield secret scan repo .  0.26s user 0.11s system 50% cpu 0.731 total
```
Otherwise we have to completely consume `commits_batch` which can take a lot of time.

Reply:
Your version also has better performance. I've added a local variable to not call `future.exception()` twice.